### PR TITLE
feat(oracle): github issue/pr verificaiton logic

### PR DESCRIPTION
**Pull request by 0xpatrickdev**
image
image

### CODE REVIEW

**Comment by dckc** [comment]
typo: verificaiton
mostly as evidence that I'm reading closely :)

File: oracle-server/src/lib/octokit.ts [hunk 1]
```diff
@@ -24,3 +43,86 @@ export function getOctokit() {
...
+      closingIssuesReferences (first:1) {
+        nodes {
+          number
+          url
+          assignees (first:1) {
+            nodes {
+              resourcePath
+              name
+            }
+          }
```
Comment (dckc) [inline oracle-server/src/lib/octokit.ts:115]: why no `url` for `assignees`?

File: web/src/components/ClaimBountyForm.tsx [hunk 2]
```diff
@@ -13,13 +19,28 @@ const ClaimBountyForm = ({ title, description }: ClaimBountyFormProps) => {
   const walletInputRef = useRef<HTMLInputElement>(null);
   const { walletAddress } = useWallet();
 
+  const mutation = useMutation<ClaimBountyOutput, Error, ClaimBountyInput>({
+    mutationFn: claimBounty,
+    onSuccess: (data) => {
+      // Handle success
+      console.log("success", data);
+      toast.success("Bounty claimed successfully!");
```
Comment (0xpatrickdev) [inline web/src/components/ClaimBountyForm.tsx:27]: todo: listen for invitation, build tx, and prompt user to sign keplr tx

File: oracle-server/src/lib/octokit.ts (hunk 1 above)
Comment (0xpatrickdev) [reply oracle-server/src/lib/octokit.ts:115]: resourcePath is `/${username}`, so this was easier to parse

Comment (0xpatrickdev) [reply oracle-server/src/lib/octokit.ts:115]: Ideally, we use `id`. There may be a way to get it via a relation but didn't did too hard

Comment (dckc) [reply oracle-server/src/lib/octokit.ts:115]: hm. if we use `url` for both PR author and issue assignee, there's no need to parse anything, right?

**Review by dckc — APPROVED** [review]
I'm kicking the tires; it seems to work pretty well.

For example, I tried to claim based on one of your PRs, and I got:

image

```
{"level":30,"time":1700175686110,"pid":2988393,"hostname":"bldbox","reqId":"req-3","req":{"method":"POST","url":"/job/claim","hostname":"localhost:5000","remoteAddress":"127.0.0.1","remotePort":54794},"msg":"incoming request"}
{"level":30,"time":1700175686725,"pid":2988393,"hostname":"bldbox","reqId":"req-3","res":{"statusCode":403},"responseTime":614.2947220206261,"msg":"request completed"}
```

**Pull request merged**